vispy/glsl/misc/spatial-filters.frag

```diff
@@ -15,208 +15,208 @@ float unpack(vec4 rgba) {
     return sign * exp2(exponent) * (mantissa * exp2(-23.0));
 }
 
-float unpack_interpolate(sampler2D kernel, vec2 uv) {
+float unpack_interpolate(sampler2D kernel, vec2 uv, vec2 pixel) {
     float u = uv.x;
     float v = uv.y;
-    float u1 = ceil(u);
-    float uf = u1 - u;
-    float d0 = unpack(texture2D(kernel, vec2(u1 - 1, v)));
-    float d1 = unpack(texture2D(kernel, vec2(u1, v)));
+    float uf = fract(u / pixel.x);
```

Comment:
this is the key, we want the fractional distance in (horizontal) pixels of the texture. `u` is normalized, i.e. on the scale 0->1. But what we want are integer indices, and we want to know how far between two indices we are. This gets us there.